libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
Since they share one queue, in theory prefetch workers could end up servicing on-demand requests, and once those are all serviced, an on-demand worker would end up servicing the prefetch requests that originally caused work to go on the prefetch work channel.  Is that right?

I guess it's not really a problem except that it if there were N prefecth requests that happened concurrently with a bunch of on-demand requests, they might end up lighting up N on-demand workers, which would tie them up for future on-demand requests and also cause more background CPU/network usage than we intended.

It might not even be that rare, since on-demand requests usually trigger the prefetch requests, and so if there's a batch of on-demand requests that's greater than the overall size of the on-demand worker pool, we could get into this situation pretty easily.

It's probably not worth worrying about, but I think it's worth a comment somewhere to remind ourselves that we're aware of it.  I guess to fix it you'd need two different queues and the ability to atomically move requests from one to the other when the priority changes.